crates/install-wheel-rs/src/wheel.rs

```diff
@@ -114,8 +114,40 @@ fn copy_and_hash(reader: &mut impl Read, writer: &mut impl Write) -> io::Result<
     ))
 }
 
-fn get_shebang(python_executable: impl AsRef<Path>) -> String {
-    format!("#!{}", python_executable.as_ref().simplified().display())
+/// Format the shebang for a given Python executable.
+///
+/// Like pip, if a shebang is non-simple (too long or contains spaces), we use `/bin/sh` as the
+/// executable.
+///
+/// See: <https://github.com/pypa/pip/blob/0ad4c94be74cc24874c6feb5bb3c2152c398a18e/src/pip/_vendor/distlib/scripts.py#L136-L165>
+fn format_shebang(layout: &Layout) -> String {
+    // Convert the executable to a simplified path.
+    let executable = layout.sys_executable.simplified_display().to_string();
+
+    // Validate the shebang.
+    if layout.os_name == "posix" {
+        // The length of the full line: the shebang, plus the leading `#` and `!`, and a trailing
+        // newline.
+        let shebang_length = 2 + executable.len() + 1;
+
+        // Apparently, Darwin can handle shebangs up to 512 characters.
+        let max_shebang_length = if layout.sys_platform == "darwin" {
+            512
+        } else {
+            127
+        };
```

Comment:
If we don't really care, we could remove the platform and OS casing. (I assume this is _safe_ to do on Windows, just not required?)

Reply:
I would gently nudge toward using just one value at first and see if that causes any issues. I assume shebangs longer than 127 are pretty rare?

Reply:
Done!